content/building-blocks/automate-and-execute-your-work/reproducible-work/google_cloud_docker.md

```diff
@@ -0,0 +1,312 @@
+
+{{% /warning %}}
+
+With this command Docker will build the images and its context, being the latter comprised of all the files located in the same path as your dockerfile. As a result of this it is generally advised to place your dockerfiles in a directory with just the files it requires in order to ensure the building process is as efficient as posible. If you want to learn more about the build command and its options you can visit [this site](https://docs.docker.com/engine/reference/commandline/build/).
+
+{{% tip %}}
+ 
+You can  assign a name (i.e. tag) to your new Docker image by addign to the code above the flag "-t" followed by the name you want to assign in between of the build command and the dot at the end as shown in the code block below. 
+
+Don't worry if you don't include this information as in that case docker will automatically assign a name to the image. However specifying a name is advisable as it will allow you to identify your images more easily.
+
+You can view a list of all your docker images in your Docker Desktop dashboard or by typing `docker images` in the command line.
+
+{{% /tip %}}
+
+{{% codeblock %}}
+```bash
+# Build a Docker image and assign a name to it
+$ docker build -t <your-image-name> .
+```
+{{% /codeblock %}}
+
+### Step 3: Docker compose
+
+Docker compose is a tool within the Docker ecosytem whose main functionality is to coordinate your containers and provide them with the services required to run smoothly. These services include actions such as communication between containers or starting up containers which require aditional instructions to do so. Although this can sound a bit confusing if you are new to Docker (You can learn more about Docker compose [here](https://docs.docker.com/compose/)), the key takeaway is that it will assist and simplify the task of replicating your environment by providing some extra information to Docker about how to do so. Additionally it is worth noting that we could perform these actions manually employing the `docker run` command, however in this way the process not as straightforward.
+
+In the same way that you needed a dockerfile to provide Docker with the necessary instructions to build an image, a docker-compose file is required to tell docker how to provide these services to a (series of) container(s).
+
+{{% tip %}}
+
+Docker compose files must be YAML files, so make sure that your docker compose file is denoted as such with the extension ".yml" 
+
+{{% /tip %}}
+
+Now it is time to upload your Docker-compose in the same way you did it with your dockerfile, and then and excute the following code in the directory where your docker-compose file is located:
+
+{{% codeblock %}}
+```bash
+# Build the docker image
+$ docker compose up 
+```
+{{% /codeblock %}}
+
+{{% warning %}}
+
+If you receive an error message after executing the codeblock above, run `sudo chmod 666 /var/run/docker.sock` and try it again.

```

Comment:
This error is probably related to not adding your username to the docker grp post-install (check comment I made above). chmod 666 gives everyone permission to mess around with the docker engine, which might not be the best idea. Could you check if adding your user to the docker grp solves this problem actually?

Reply:
That was indeed the case, after adding my username to the group this issue was solve. Consequentiall I have deleted this warning from the building block.